apps/storeutl.c

```diff
@@ -157,7 +164,7 @@ int storeutl_main(int argc, char *argv[])
                            prog);
                 goto end;
             }
-            subject = parse_name(opt_arg(), MBSTRING_UTF8, 1, "subject");
+            subject = parse_name(opt_arg(), MBSTRING_UTF8, multirdn, "subject");
```

Comment:
I guess that @levitte did not provide that option for `storeutl` simply to save some the little effort implementing it and had it on unconditionally just in case someone needs this capability.
In all other apps where the `-multivalue-rdn` was already present it is off by default. The need to switch it on should be pretty rare (as mentioned in the lib documentation, the whole multi-RDN business is very seldomly used).
Moreover, for binary options we do not have the possibility to have it on by default and while offering to switch it off (i.e., we do not have something like`-multivalue-rdn=0`) on the command line; we could only have, e.g., `-no-multivalue-rdn`, which again would be inconsistent with the other apps.
For these reason I'd like to stick with `-multivalue-rdn` off by default.

Reply:
Quite frankly, I don't quite see why it couldn't be permanently on.  Please remember that OpenSSL has a lot of history, and these kind of flags are there *in other commands* to ensure that the user didn't get sudden surprises in terms of output with newer OpenSSL versions, after output variants where implemented.

`storeutl` is a relatively new command and thereby doesn't have historically different behaviour of that sort...

Also, especially since multi-valued RDNs are, according to you, fairly rare, this flag doesn't matter in the majority of the cases...  that makes the option even more pointless, if I may be so bold.

Reply:
Good point.
So for the new apps `storeutl` and `cmp` we could unconditionally enable support for multi-valued RDNs.
The only problem I see with this is that unsuspecting (unescaped) usage of the `+` character in RDN values would have unexpected effects.

Reply:
I think having commands with different behavior for the same thing (parsing X509 names) is a bad mistake.

Reply:
Yes, consistency is a good point in the direction currently done in this PR.

A third possibility would be to simply remove all `-multivalue-rdn` options,
maybe adding a warning in case an unescaped `+` is used within an RDN value.

Reply:
I've meanwhile implemented the idea I have given 5 days back:
* reverted adding `-multivalue-rdn` options
* made existing ones obsolete
* add a hint in case a parse error may be due to unsuspecting use of `+` in value strings.

Reply:
So all apps that parse Distinguished Names now have multi-RDN support enabled unconditionally.
I've also reflected this (and the obsoletion of `-multivalue-rdn`) in the documentation.